File: src/main/java/com/jewelry/pos/service/PersonalPersonService.java
package com.jewelry.pos.service;

import com.jewelry.pos.domain.entity.PersonalPerson;
import com.jewelry.pos.domain.repository.PersonalPersonRepository;
import com.jewelry.pos.web.dto.PersonalPersonDTO;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

@Service
@RequiredArgsConstructor
public class PersonalPersonService {

    private final PersonalPersonRepository personalPersonRepository;

    @Transactional
    public PersonalPerson createPerson(PersonalPersonDTO dto) {
        PersonalPerson person = new PersonalPerson();
        person.setName(dto.name());
        person.setPhoneNumber(dto.phoneNumber());
        person.setAddress(dto.address());
        person.setNotes(dto.notes());
        return personalPersonRepository.save(person);
    }

    @Transactional
    public PersonalPerson updatePerson(String id, PersonalPersonDTO dto) {
        PersonalPerson person = personalPersonRepository.findById(id)
                .orElseThrow(() -> new IllegalArgumentException("Person not found"));
        person.setName(dto.name());
        person.setPhoneNumber(dto.phoneNumber());
        person.setAddress(dto.address());
        person.setNotes(dto.notes());
        return personalPersonRepository.save(person);
    }

    @Transactional(readOnly = true)
    public List<PersonalPerson> getAllPersons() {
        return personalPersonRepository.findAll();
    }

    @Transactional(readOnly = true)
    public PersonalPerson getPersonById(String id) {
        return personalPersonRepository.findById(id)
                .orElseThrow(() -> new IllegalArgumentException("Person not found"));
    }

    @Transactional
    public void deletePerson(String id) {
        personalPersonRepository.deleteById(id);
    }
}

File: frontend/src/app/core/services/personal-person.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface PersonalPerson {
  id: string;
  name: string;
  phoneNumber?: string;
  address?: string;
  notes?: string;
}

export interface PersonalPersonRequest {
  name: string;
  phoneNumber?: string;
  address?: string;
  notes?: string;
}

@Injectable({
  providedIn: 'root'
})
export class PersonalPersonService {
  private http = inject(HttpClient);
  private readonly API_URL = '/api/personal-persons';

  getAllPersons(): Observable<PersonalPerson[]> {
    return this.http.get<PersonalPerson[]>(this.API_URL);
  }

  getPersonById(id: string): Observable<PersonalPerson> {
    return this.http.get<PersonalPerson>(`${this.API_URL}/${id}`);
  }

  createPerson(request: PersonalPersonRequest): Observable<PersonalPerson> {
    return this.http.post<PersonalPerson>(this.API_URL, request);
  }

  updatePerson(id: string, request: PersonalPersonRequest): Observable<PersonalPerson> {
    return this.http.put<PersonalPerson>(`${this.API_URL}/${id}`, request);
  }

  deletePerson(id: string): Observable<void> {
    return this.http.delete<void>(`${this.API_URL}/${id}`);
  }
}
